src/Chainweb/Cut/Test.hs

```diff
@@ -143,9 +170,27 @@ createNewCut
     -> cid
     -> Cut
     -> Either MineFailure (T2 BlockHeader Cut)
-createNewCut n t pay i c = do
+createNewCut = createNewCut' checkPOW
+  where
+    checkPOW h = checkTarget (_blockTarget h) $ _blockPow h
+
+-- | Create a new block. Only produces a new cut but doesn't insert it into the
+-- chain database. Optional POW validation.
+--
+createNewCut'
+    :: HasCallStack
+    => HasChainId cid
+    => (BlockHeader -> Bool)
+    -- ^ POW validator
```

Comment:
It's possible, but that's not the intention. The point here is to use mocking techniques to avoid violating separation-of-concerns in unit tests.

Using `ChainwebVersion` is less desirable, as it is overloaded with three separate concerns:
1. A protocol version. This in itself (protocol version compliance) is not of interest to many subsystems.
2. A graph. This is relevant to multi-chain-aware subsystems.
3. A hashing algorithm/DA/difficulty regime. This is relevant to mining- and DA-concerned subsystems.

Notice how all three of these properties are _irrelevant_ to some single-chain subsystem that is not involved in mining or DA or protocols. 

Here, by mocking out the nonce validation, a test is able to precisely specify whether #3 is of concern. `(const True)` (maybe with a suitably literate type synonym) is the most accurate way to say that mining difficulty is irrelevant; in this case it is strictly better as it makes a test less brittle if later on it needs to change something regarding #2 or #1. 

Finally, leaving it to `ChainwebVersion` means that the only true safe way to use these functions is via a `mine` function, _in case_ the protocol requires mining. This necessarily involves a clock, randomness etc, and is specifically what I wanted to avoid.

Reply:
Short reply:

the test already depends on being used with a chainweb version that uses a trivial POW target, because `insert` does POW validation. Passing `const True` seems thus redundant.

Long reply:

The consistency rules for a chainweb are part of the chainweb version, which are implemented by the validation rules. Validation includes:

1. block header validation (intrinsic and indicutive)
2. cut validation (braiding structure of chainweb)
3. payload validation (construction of payload Merkle tree, tx uniqueness, tx timing validation, gas, etc.)
4. pact validation (computation of outputs via pact validation)

In the original design we used parameterized test chainweb versions to tweak consistency properties for testing. In most cases properties are statically bound to a particular version. Currently only the graph can be changed freely on some test versions. But @fosskers and I have been discussing to add free more parameters, like for instance the block rate.

In the current code we don't follow this original design everywhere, but we use chainweb versions to disable certain validation properties. For instance most test versions already use a trivial POW target, which is equivalent to passing `const True` to the POW validator.

Reply:
The idea behind using the chainweb version to relax validation constraints is to gather all configurations in a single place (`Chainweb.Version`) and avoid accidentally disabling or changing consistency properties. We try to make it impossible (or at least hard) to run code with chainweb version `Mainnet01` (or any other production version) with some consistency rules altered or disabled.

Reply:
Hmm, I got confused by the check in the code here, I didn't realize `insert` did it as well.

In that case the indirection is indeed unnecessary. On the larger topic above, I might want to allow tests to set the target to `0` (irrespective of what comes out of `newHeader` and it's merkle hash), but that's for a later debate. I'll switch to using `createNewCut` and drop this function.